legacy_tests/spi_master_sync_clock_port_sharing/spi_master_sync_clock_port_sharing.xc

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020, XMOS Ltd, All rights reserved
+#include <platform.h>
+#include <xclib.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <print.h>
+#include "spi.h"
+#include "spi_sync_tester.h"
+
+in buffered port:32    p_miso   = XS1_PORT_1A;
+out port               p_ss[1]  = {XS1_PORT_1B};
+out buffered port:32   p_sclk   = XS1_PORT_1C;
+out buffered port:32   p_mosi   = XS1_PORT_1D;
+clock                  cb       = XS1_CLKBLK_1;
+
+out port setup_strobe_port = XS1_PORT_1E;
+out port setup_data_port = XS1_PORT_16B;
+
+[[combinable]]
+void app(client interface spi_master_if spi_i, int mosi_enabled, int miso_enabled){
+    int count = 0;
+    timer always;
+    int start;
+    always :> start;
+
+    while (1) {
+        select {
+            case always when timerafter(start) :> void:
+                unsigned inter_frame_gap = 1000;
+                unsigned device_id = 0;
+                spi_mode_t mode = SPI_MODE_0;
+                unsigned speed_in_kbps = 500;
+
+                // do a test transfer
+                int ret = test_transfer8(spi_i, setup_strobe_port, setup_data_port, device_id,
+                                         inter_frame_gap, mode, speed_in_kbps,
+                                         mosi_enabled, miso_enabled);
+                if (ret != 0)
+                    printf("Error %d in round %d\n", ret, count);
+
+                // now do something to the clock port
+                {   unsigned sclk;
+                    asm volatile("ldw %0, dp[p_sclk]" : "=r"(sclk));
+                    asm volatile("setclk res[%0], %1" :: "r"(sclk), "r"(XS1_CLKBLK_REF));
+                }
+
+                count++;
+                if (count == 2)
```

Comment:
Doing the test just twice seems low. Does it take a long time or something?

Reply:
No particular reason. I can make it 8. It takes 1.5 minutes to do two iterations and 5.5 minutes to do eight iterations. That should be an insignificant cost given that we can expect the SPI library to change only infrequently.